### ISSUE TITLE
Support for LILYGO T4 S3

### ISSUE THREAD

**Issue by il0vew33d** [jgauchia/IceNav-v3#260]
please add LILYGO T4 S3. It has a 450x600 AMOLED touchscreen.

https://lilygo.cc/products/t4-s3

thanks a lot.

**Comment by jgauchia** [comment]
Hi,
I'll try to add support for LilyGo T4 S3. I don't have this board, but I'll try it

**Comment by jgauchia** [comment]
I'll try to give support for this board in the Next release, please be patient and stay tunned

**Comment by jgauchia** [comment]
I've waiting for a T4 S3 Board, stay tunned for new updates to bring support for it

**Comment by jgauchia** [comment]
Hi @il0vew33d , I’m waiting to receive a T4-S3 shortly to provide support as you requested. I just wanted to let you know that your issue is on my radar, and as soon as I receive the board, I'll get to work on it. I’d also like to mention that over the past few days, I’ve been looking into how it works, etc., and noticed that it’s not currently supported by the graphics library I use. That shouldn’t be a problem—on the contrary, it’s a challenge. Plus, I’m interested in providing this support since it has a lot of potential. I’ll keep you updated on my progress if you don’t mind, and I might ask you to validate some aspects of my firmware as I go along

**Comment by il0vew33d** [comment]
hi jordi,
that's great news. what do you mean with graphics library? T4-S4 supports LVGL, I thought you also use LVGL?
https://github.com/Xinyuan-LilyGO/LilyGo-AMOLED-Series

of course  we can discuss some aspects of your firmware.
frank. :-)

**Comment by jgauchia** [comment]
Yes, T4-S3 supports LVGL, but my project is based in LovyanGFX for all graphics, etc... This isn't a problem,I just ask you for a little patience; I’m interested in supporting as many boards as possible in my humble project hahaha.

**Comment by jgauchia** [comment]
But don’t worry, with a bit of time, etc., I’ll add support. I’m interested in it

**Comment by il0vew33d** [comment]
ah, ok. now I understand. please don't stress, we have time....

**Comment by jgauchia** [comment]
hahahah, no problem... Quite the opposite, I’d like to make my firmware support different boards... of course, with the dedication I can give, as it’s a personal and/or hobby project

**Comment by jgauchia** [comment]
hi @il0vew33d , I'm waiting to receive today a T4-S3.... I'm expecting to give support for this board... stay tunned...
Thank you

**Comment by jgauchia** [comment]
![image](https://github.com/user-attachments/assets/5314640c-60f0-4129-b0a2-be8c57dfadf5)

Well, I've one... I'm goingo to give support, stay tunned.

**Comment by jgauchia** [comment]
Hi @il0vew33d , I'm working on it...
There are some issues with this board and LovyanGfx graphics library than I use.

Open a discussion with them: https://github.com/lovyan03/LovyanGFX/discussions/663#discussioncomment-11711098

**Comment by jgauchia** [comment]
Hi @il0vew33d , lovyangfx has a new branch to give support (limited) to T4-S3 .
I'm going to test it and try to give support to It in Icenav

**Comment by jgauchia** [comment]
Some progress...

<img src="https://github.com/user-attachments/assets/5f5fe05d-b2e1-4b2f-88cc-4f047bcdb547" width=20% height=20%>

<img src="https://github.com/user-attachments/assets/9401f342-6d32-48fd-a328-bb7aefca16f0" width=20% height=20%>

**Comment by il0vew33d** [comment]
wouw, cool. that's good news. :-)

meanwhile i also have a lilygo t-deck plus and installed icenav. is there any description how it works?
